octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+                           f"Error canceling order (id {order_id}), the order is still uncanceled{cancel_resp_message}"
+            else:
+                # Order is not found: it has successfully been cancelled 
+                # (some exchanges don't allow to get a cancelled order).
+                success = True
+                return success, None
+        except ccxt.OrderNotFound:
+            # Order is not found: it has successfully been cancelled 
+            # (some exchanges don't allow to get a cancelled order).
+            success = True
+            return success, None
+
+    async def cancel_stop_order_using_stop_loss_endpoint(self, order_id: str, symbol: str = None,
+                                                         **kwargs: dict) -> typing.Tuple[bool, str or None]:
+        # some exchange have an extra stop/take profit endpoint
+        # from bybit docs: You may cancel all untriggered conditional orders or take profit/stop loss order.
+        # Essentially, after a conditional order is triggered, it will become an active order. So, when a conditional
+        # order is triggered, cancellation has to be done through the active order endpoint for any unfilled or
+        # partially filled active order
+        kwargs["stop_order_id"] = order_id
+        return await self.cancel_order_default(order_id, symbol=symbol, kwargs=kwargs)
 
     async def get_positions(self, **kwargs: dict) -> list:
-        return await self.client.fetch_positions(params=kwargs)
+        """
+            all known get_positions methods should be added here
+            so untested exchanges have higher chance of success
+        """
+        defined_methods = self.connector_config.GET_POSITION_METHODS
+        positions = []
+        if self.get_positions_linear.__name__ in defined_methods:
+            positions += await self.get_positions_linear(**kwargs)
+        if self.get_positions_inverse.__name__ in defined_methods:
+            positions += await self.get_positions_inverse(**kwargs)
+        if self.get_positions_swap.__name__ in defined_methods:
+            positions += await self.get_positions_swap(**kwargs)
+        if self.get_positions_option.__name__ in defined_methods:
+            positions += await self.get_positions_option(**kwargs)
+        if self.get_positions_with_private_get_position_risk.__name__ in defined_methods:
+            positions += await self.get_positions_with_private_get_position_risk()
+        return positions
+
+    async def get_positions_with_private_get_position_risk(self) -> list:
+        try:
+            if self.client.has.get("fapiPrivate_get_positionrisk"):
+                return await self.exchange_manager.exchange.parse_positions(
+                    await self.client.fapiPrivate_get_positionrisk())
+        except Exception as e:
+            self.logger.exception(e, True, f"Failed to load positions using private_get_position_risk")
+        return []
```

Comment:
I don't think we need this method. CCXT private method are exchange specific, here we are in the generic part.

Reply:
I removed it